### DISCUSSION TITLE
Request-Rate Exceeded alchemy-wagmi

### DISCUSSION THREAD

**Discussion by Irwinhrdz** [wevm/wagmi#1142] [General]
Hello mates , I don't know if this is the right place for this, but the restriction comes from Wagmi and Alchemy. I am in the process of developing an app with nextjs and wagmi, I have been developing it for a couple of months without provider problems, now it seems I have an error with the PublicProvider because it exceeded the number of requests (according to the image), I have already changed the provider by AlchemyProvider and I have put my own keys and the error does not come out but I do not get any data from the SmartContract, I have put jsonRpcProvider as the second option with https://goerli.infura.io/v3/ and my own address and with the parameters of Priority that wagmi mentions in the documentation, the error no longer appears but it still does not bring me contract data... could someone help me? How can I make my app consume the contract again and which provider should I use?

**Comment by mongchanghsi** [comment]
Similar question, I am quite concerned that the CU in alchemy may exceed due to the numerous calls coming from rainbowkit. I am wondering if the following code snippets works as a form of secondary alchemyProvider? With the same priority level, I am getting it to even-out between the two providers so it will not be easily hit the limit of Alchemy CU

```
const { chains, provider } = configureChains(
  [chain.mainnet],
  [
    alchemyProvider({ apiKey: envConfig.ALCHEMY_API_KEY, priority: 0 }),
    alchemyProvider({ apiKey: envConfig.ALCHEMY_API_KEY_SUB, priority: 0 }),
    publicProvider({ priority: 1 }),
  ]
);
```

**Reply by Irwinhrdz** [reply]
Remove PublicProvider, change alchemy provider api keys to new ones, rollback wagmi version to 0.6.8, and everything is back to normal... that worked for me

**Reply by mongchanghsi** [reply]
My main concern being when you are in a situation where you can't easily redeploy a new api key because your event is a short period of time. Say a 10 minutes event with thousands of people connecting to your website at once. That will definitely shot up the calls to Alchemy. If the limit is hit, and people are facing issue due to this rate limit, people will definitely lose trust in the project. I had one experience using Infura with a limit of 100,000 and it hit within 2 minutes, causing people facing issues on the website, which led to people losing trust in the project due to the tech being faulty.
